SU-HSDO/suhumsci
# Feat(STN-896): add js linting

# READY FOR REVIEW

## Summary
Identified as part of the strategy to increase stability & confidence in the application, JavaScript linting needed to be added. With this PR we achieve:

- linting for `humsci_basic`
- Webpack upgrades
- resolution of newly discovered lint errors.

## Need Review By (Date)
_asap_

## Urgency
medium

## Steps to Test
1. run `npm i` from `docroot/themes/humsci/humsci_basic`
2. run `npm run start` to confirm base theme builds as expected
3. run `npm run lint:js` to confirm eslint task runs independently of ☝️. You may need to edit a javaScript file to cause an error.
4. With an error present in your JS, once again run `npm run start` to confirm that a lint error causes the build to fail. Functionally speaking, all the code should work exactly the same as it did prior to the PR.
5. Switch between themes and test between different components that rely upon javaScript.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
